checker/src/org/checkerframework/checker/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,653 @@
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.ImplicitsTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.PropagationTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.framework.util.AnnotatedTypes;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.Pair;
+
+/**
+ *  Implements the introduction rules for the Lower Bound Checker.
+ *  <pre>
+ *  The type hierarchy is:
+ *
+ *  Top = lbu ("Lower Bound Unknown")
+ *   |
+ *  gte-1 ("Greater than or equal to -1")
+ *   |
+ *  nn  ("NonNegative")
+ *   |
+ *  pos ("Positive")
+ *  </pre>
+ *  In general, check whether the constant Value Checker can determine the
+ *  value of a variable; if it can, use that; if not, use more specific rules
+ *  based on expression type. These rules are documented on the functions
+ *  implementing them.
+ */
+public class LowerBoundAnnotatedTypeFactory
+        extends GenericAnnotatedTypeFactory<
+                CFValue, CFStore, LowerBoundTransfer, LowerBoundAnalysis> {
+
+    /** The canonical @GTENegativeOne annotation. */
+    public final AnnotationMirror GTEN1 = AnnotationUtils.fromClass(elements, GTENegativeOne.class);
+    /** The canonical @Negative annotation. */
+    public final AnnotationMirror NN = AnnotationUtils.fromClass(elements, NonNegative.class);
+    /** The canonical @Positive annotation. */
+    public final AnnotationMirror POS = AnnotationUtils.fromClass(elements, Positive.class);
+    /** The canonical @LowerBoundUnknown annotation. */
+    public final AnnotationMirror UNKNOWN =
+            AnnotationUtils.fromClass(elements, LowerBoundUnknown.class);
+
+    /**
+     * Provides a way to query the Constant Value Checker, which computes the
+     * values of expressions known at compile time (constant prop + folding).
+     */
+    private final ValueAnnotatedTypeFactory valueAnnotatedTypeFactory =
+            getTypeFactoryOfSubchecker(ValueChecker.class);
+
+    public LowerBoundAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        this.postInit();
+    }
+
+    @Override
+    protected LowerBoundAnalysis createFlowAnalysis(
+            List<Pair<VariableElement, CFValue>> fieldValues) {
+        return new LowerBoundAnalysis(checker, this, fieldValues);
+    }
+
+    /**
+     * Hack to make postfix increments and decrements work correctly.
+     * This is a workaround for issue #867:
+     * https://github.com/typetools/checker-framework/issues/867
```

Comment:
I just committed a general workaround for this issue (1d0f8c49604e04f17e5e6677aa20ac4c9693a9a9), so you should be able to remove this hack now.